src/com/google/javascript/jscomp/PolymerProtectStaticProperties.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2017 The Closure Compiler Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.javascript.jscomp;
+
+import com.google.javascript.rhino.IR;
+import com.google.javascript.rhino.Node;
+
+/**
+ * Prevents renaming of the Polymer 2 static properties "is" and "config" by adding externs to the
+ * Function.prototype.
+ *
+ * <p>This pass must run after type checking since otherwise all functions would have an "is" and
+ * "config" property.
+ */
+class PolymerProtectStaticProperties implements CompilerPass {
+
+  private final AbstractCompiler compiler;
+
+  PolymerProtectStaticProperties(AbstractCompiler compiler) {
+    this.compiler = compiler;
+  }
+
+  @Override
+  public void process(Node externs, Node root) {
+    FindPolymerExterns findPolymerExterns = new FindPolymerExterns();
+    NodeTraversal.traverseEs6(compiler, externs, findPolymerExterns);
+
+    if (findPolymerExterns.hasPolymerElementExterns()) {
+      addExtern("is");
```

Comment:
No - it's not. With the class based style, the compiler does MUCH better at understanding the syntax and fully enables optimizations. I think the real answer is https://github.com/google/closure-compiler/issues/1693#issuecomment-303981793 but I don't know what it would take to land such a change at Google.